crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1070 @@
+///   And it is not included in Typst.
+/// - [Parula](https://www.mathworks.com/help/matlab/ref/parula.html)
+///   is a good color map included in matlab, but it is not included as
+///   a preset in Typst. This is because it is owned by MathWorks and is not public.
+///
+#[ty(scope)]
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub enum Gradient {
+    Linear(LinearGradient),
+}
+
+impl Debug for Gradient {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Linear(linear) => linear.fmt(f),
+        }
+    }
+}
+
+#[scope]
+impl Gradient {
+    pub const SPECTRAL: fn() -> Array = spectral;
+    pub const VIRIDIS: fn() -> Array = viridis;
+    pub const INFERNO: fn() -> Array = inferno;
+    pub const MAGMA: fn() -> Array = magma;
+    pub const PLASMA: fn() -> Array = plasma;
+    pub const ROCKET: fn() -> Array = rocket;
+    pub const MAKO: fn() -> Array = mako;
+    pub const VLAG: fn() -> Array = vlag;
+    pub const ICEFIRE: fn() -> Array = icefire;
+    pub const FLARE: fn() -> Array = flare;
+    pub const CREST: fn() -> Array = crest;
+
+    /// Creates a new linear gradient.
+    #[func(title = "Linear gradient")]
+    pub fn linear(
+        /// The call site of this function.
+        span: Span,
+
+        /// The color stops of the gradient.
+        #[variadic]
+        stops: Vec<Spanned<Stop>>,
+
+        /// The direction or angle of the gradient.
+        #[named]
+        #[default(DirOrAngle::Dir(Dir::LTR))]
+        dir: DirOrAngle,
+
+        /// The color space in which to interpolate the gradient.
+        ///
+        /// Defaults to a perceptually uniform color space called
+        /// [Oklab]($color.oklab).
+        #[named]
+        #[default(ColorSpace::Oklab)]
+        space: ColorSpace,
+
+        /// The relative placement of the gradient.
+        ///
+        /// - `"this"`: The gradient is relative to the bounding box of the
+        ///   container onto which it is painted.
+        /// - `"parent"`: The gradient is relative to the bounding box of the
+        ///   parent that contains the element onto which the gradient is applied.
+        ///
+        /// For an element placed at the root/top level of the document, the parent
+        /// is the page itself. For other elements, the parent is the first block or
+        /// box that contains the element.
+        #[named]
+        #[default(Smart::Auto)]
+        relative: Smart<Relative>,
+    ) -> SourceResult<Gradient> {
+        if stops.len() < 2 {
+            bail!(error!(span, "a gradient must have at least two stops")
+                .with_hint("try filling the shape with a single color instead"));
+        }
+
+        let stops = process_stops(&stops)?;
+
+        Ok(Self::Linear(LinearGradient {
+            stops,
+            angle: dir.into(),
+            space,
+            relative,
+            anti_alias: true,
+        }))
+    }
+
+    /// Return the stops of this gradient.
```

Comment:
For consistency with the other functions
```suggestion
    /// Returns the stops of this gradient.
```